### PR TITLE
More posix (pid) range lock tests

### PR DESCRIPTION
**Pull request by sporksmith**
Progress on #2258

### CODE REVIEW

File: src/test/file_locks/test_file_locks.rs [hunk 1]
```diff
@@ -246,6 +249,455 @@ fn test_contested_pid_locks(cmd: FcntlPosixSetlkUncontestedCommand) -> anyhow::R
...
+            fd: file.as_raw_fd(),
+            cmd: cmd.into(),
+            flock: wr_flock,
+        })?
+        .to_result()?;
+
+    // Double-check that we see the lock.
+    {
+        let out_flock = fcntl_lock(file.as_raw_fd(), FcntlFlockCommand::F_GETLK, &wr_flock)?;
+        ensure_ord!(out_flock, ==, libc::flock {l_pid: child.pid(), ..wr_flock });
```
Comment (robgjansen) [inline src/test/file_locks/test_file_locks.rs:386]: Nit: consider unnesting.

File: src/test/file_locks/test_file_locks.rs [hunk 2]
```diff
@@ -246,6 +249,455 @@ fn test_contested_pid_locks(cmd: FcntlPosixSetlkUncontestedCommand) -> anyhow::R
...
+        ensure_ord!(out_flock.l_pid, ==, child.pid());
+    }
+
+    // Have child exit, which should cause the lock to be released.
+    drop(child);
+
+    // We should no longer see the lock.
+    {
+        let out_flock = fcntl_lock(file.as_raw_fd(), FcntlFlockCommand::F_GETLK, &wr_flock)?;
+        ensure_ord!(out_flock, ==, libc::flock {l_type: libc::F_UNLCK.try_into().unwrap(), ..wr_flock });
```
Comment (robgjansen) [inline src/test/file_locks/test_file_locks.rs:396]: Nit: consider unnesting.

File: src/test/file_locks/test_file_locks.rs [hunk 3]
```diff
@@ -246,6 +249,455 @@ fn test_contested_pid_locks(cmd: FcntlPosixSetlkUncontestedCommand) -> anyhow::R
...
+    // dropped as a result of tearing down the thread.
+    let mut child = ForkedChild::new(handle_lock_request)?;
+    let out_flock = child
+        .send_recv(&SerializedLockRequest {
+            fd,
+            cmd: FcntlFlockCommand::F_GETLK.into(),
+            flock: wr_flock,
+        })?
+        .to_result()?;
+    ensure_ord!(out_flock, ==, libc::flock {
```
Comment (robgjansen) [inline src/test/file_locks/test_file_locks.rs:431]: Nit: consider unnesting.

File: src/test/file_locks/test_file_locks.rs [hunk 4]
```diff
@@ -327,6 +779,46 @@ fn test_zero_len_pid_locks(cmd: FcntlPosixSetlkUncontestedCommand) -> anyhow::Re
...
+    // Use GETLK from child to inspect the lock.
+    {
+        let out_flock = child
+            .send_recv(&SerializedLockRequest {
+                fd: file.as_raw_fd(),
+                cmd: FcntlFlockCommand::F_GETLK.into(),
+                flock: negative_len_flock,
+            })?
+            .to_result()?;
+        ensure_ord!(out_flock, ==, libc::flock {
```
Comment (robgjansen) [inline src/test/file_locks/test_file_locks.rs:810]: Nit: consider unnesting.

**Review by robgjansen — APPROVED** [review]
The tests look great! I'm surprised at the number of cases needing tests, but glad we're comprehensively adding these!

FYI, I wanted to make sure I'm not being too annoying with my requests for un-nesting the assertions, so I looked into it. You're probably already familiar, but I found out this is known as the Arrange-Act-Assert (AAA) pattern and is generally considered a best practice for test readability. Cool :)

**Comment by sporksmith** [comment]
> The tests look great! I'm surprised at the number of cases needing tests, but glad we're comprehensively adding these!

Thanks! Yeah there turn out to be a lot of "fun" / interesting corner cases to test :)

> FYI, I wanted to make sure I'm not being too annoying with my requests for un-nesting the assertions, so I looked into it. You're probably already familiar, but I found out this is known as the Arrange-Act-Assert (AAA) pattern and is generally considered a best practice for test readability. Cool :)

Yeah, I think I've seen something along the AAA pattern before. I think it's an ok suggestion as a starting point, but I'm not convinced to be dogmatic about it :)

> Nit: consider unnesting.

Based on the feedback from the previous PR I kept/moved significant function calls out of being inlined into the `ensure_eq!` invocations, but IMO readability wouldn't be improved by lifting what are mostly literal structs into a `let expected = { ... }`.

e.g. to take it a step further, I think `let expected = 3; assert_ord!(actual, ==, expected)` would be worse than `assert_ord!(actual, ==, 3)`.

For a complicated enough `expected` it might be worth assigning to a temp this way, but IMO these don't quite clear the bar
